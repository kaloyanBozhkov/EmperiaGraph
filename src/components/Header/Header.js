import React from 'react'
import Button from '../UI/Button/Button'

import styles from './styles.module.scss'

const Header = ({ activeMenu = true, toggleMenu }) => {

  return (
    <div className={styles.header}>
      <div className={styles.logo}>
        <h1>EmperiaGraph</h1>
        <h3>(v0.0.1)</h3>
      </div>
      <div className={styles.btnWrapper}>
        <Button icon={activeMenu ? 'close' : 'bars'} modifier="blank" onClick={() => toggleMenu(!activeMenu)} />
      </div>
    </div>
  )
}

export default Header
