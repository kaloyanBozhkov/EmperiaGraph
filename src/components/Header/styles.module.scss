@import 'mixins';
@import 'keyframes';
@import 'variables';

.header {
    display: flex;
    justify-content: space-between;
    height: 4rem;
    background: linear-gradient(45deg, darken($primary-color, 20%), $primary-color);
    box-shadow: 1px 1px 6px rgba(0, 0, 0, 0.6);

    .logo {
        display: flex;
        align-items: center;
        h1,
        h3 {
            color: white;
            display: flex;
            align-items: center;
        }

        h3 {
            padding: 0 .2rem;
            font-size: 0.5rem;
            display: flex;
            height: 1rem;
            margin-top: 2rem;
        }

        h1 {
            padding-left: 1rem;
        }
    }

    .btnWrapper {
        width: 3rem;
        height: 100%;
        display: flex;
        justify-content: center;
        align-items: center;

        svg {
            font-size: 1.5rem;
            color: white;
        }
    }

    @include animate(slideDown, 0.4s);
}
