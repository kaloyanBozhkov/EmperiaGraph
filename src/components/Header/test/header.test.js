import React from 'react'
import { shallow } from 'enzyme'
import Header from '../Header'

describe('tesitng header', () => {
  it('should render header component', () => {
    const wrapper = shallow(<Header />)
    expect(wrapper).toMatchSnapshot()
  })
})
