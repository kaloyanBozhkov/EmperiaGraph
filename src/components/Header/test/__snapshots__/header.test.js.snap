// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tesitng header should render header component 1`] = `
<div
  className="header"
>
  <h1>
    EmperiaGraph
  </h1>
  <h3>
    v0.0.1
  </h3>
</div>
`;
